src/attribute.ts

```diff
@@ -175,15 +180,20 @@ export class Attribute {
     child.parent = this
     this.children.push(child)
   }
-
-  // addSlice adds a slice attribute to this and update the parent of the slice with the current parent.
+  // addChoice adds a choice type attribute to this and update its parent with the current parent.
   // Note that if this attribute is an item of an array, the slice must be an item as well (and we pass along the current index)
-  addSlice(slice: Attribute) {
-    slice.parent = this.parent
+  addChoice(choice: Attribute) {
+    choice.parent = this.parent
     if (this.isItem) {
-      slice.isItem = true
-      slice.index = this.index
+      choice.isItem = true
+      choice.index = this.index
     }
```

Comment:
why did you remove this logic in `addSlice` ?

Reply:
also, I don't think a value[x] can be an array so I'd remove this in `addChoice`

Reply:
Because I don't think we need slices to have indices in an array. I saw like that:

parent
  slice1[]
  slice2[]
  slice3[]

Reply:
Hum, it might be true! I kept it because I just replaces what was called slices by choices.
Didn't you have problems without it before?